walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,125 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.*;
 
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
-  }
+    public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
+        //the time slots that mandatory attendees are unavailable
+        ArrayList<TimeRange> mandatoryAttendeesUnavailableTimes = new ArrayList<TimeRange>();
+        //the time slots that mandatory and optional attendees are unavailable
+        ArrayList<TimeRange> optionalAttendeesUnavailableTimes = new ArrayList<TimeRange>();
+        for (Event event : events) {
+            HashSet<String> intersection;
+            intersection = new HashSet<String>();
+            intersection.addAll(event.getAttendees());
+            intersection.retainAll(request.getAttendees());
+            //if a mandatory attendee has an event
+            if (intersection.size() > 0) {
+                mandatoryAttendeesUnavailableTimes.add(event.getWhen());
+                optionalAttendeesUnavailableTimes.add(event.getWhen());
+            }
+            intersection = new HashSet<String>();
+            intersection.addAll(event.getAttendees());
+            intersection.retainAll(request.getOptionalAttendees());
+            //if an optional attendee has an event
+            if (intersection.size() > 0) {
+                optionalAttendeesUnavailableTimes.add(event.getWhen());
+            }
+        }
+        //the time slots that mandatory attendees are available
```

Comment:
These comments seem to largely restate the variable names.  Comments should generally focus more on the "how" and "why" of the code than the "what".